Tasks/Common/compression-common/FileExtractor.ts

```diff
@@ -0,0 +1,179 @@
+        return CompressedFile.Zip;
+    } else if (file.endsWith('.7z')) {
+        return CompressedFile.SevenZip;
+    } else {
+        throw new Error(tl.loc('UnsupportedFileExtension'));
+    }
+}
+
+type Extractor = (file: string, destination: string) => Promise<void>;
+
+/** Extract a zip file with the 'unzip' utility. */
+async function unzipExtract(file: string, destination: string): Promise<void> {
+    console.log(tl.loc('UnzipExtractFile', file));
+
+    const unzip: tr.ToolRunner = tl.tool('unzip');
+    unzip.arg(file);
+    await unzip.exec(<tr.IExecOptions>{ cwd: destination });
+}
+
+/** Extract a zip file using PowerShell. */
+async function powershellExtract(file: string, destination: string): Promise<void> {
+    console.log(tl.loc('UnzipExtractFile', file));
+
+    // build the powershell command
+    const escapedFile = file.replace(/'/g, "''").replace(/"|\n|\r/g, ''); // double-up single quotes, remove double quotes and newlines
+    const escapedDest = destination.replace(/'/g, "''").replace(/"|\n|\r/g, '');
+    const command = `
+        $ErrorActionPreference = 'Stop';
+        try
+        {
+            Add-Type -AssemblyName System.IO.Compression.FileSystem
+        }
+        catch { };
+        [System.IO.Compression.ZipFile]::ExtractToDirectory('${escapedFile}', '${escapedDest}')`;
+
+    // change the console output code page to UTF-8
+    const chcpPath = path.join(process.env.windir, "system32", "chcp.com");
+    await tl.exec(chcpPath, '65001');
+
+    // run powershell
+    const powershell: tr.ToolRunner = tl.tool('powershell')
+        .line('-NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command')
+        .arg(command);
+    await powershell.exec();
+}
+
+/** Extract a file using 7zip. */
+const sevenZipExtract = (isWindows: boolean) => async (file: string, destination: string): Promise<void> => {
+    console.log(tl.loc('SevenZipExtractFile', file));
+
+    const sevenZipLocation = isWindows ? path.join(__dirname, '7zip/7z.exe') : tl.which('7z', true);
+    const sevenZip = tl.tool(sevenZipLocation);
+
+    sevenZip.arg('x');
+    sevenZip.arg('-o' + destination);
+    sevenZip.arg(file);
+
+    const execResult = sevenZip.execSync();
+    if (execResult.code !== tl.TaskResult.Succeeded) {
+        tl.debug('execResult: ' + JSON.stringify(execResult));
+    }
+}
+
+/* Extract a compressed tar archive using the 'tar' utiltity. */
+async function tarExtract(file: string, destination: string): Promise<void> {
+    console.log(tl.loc('TarExtractFile', file));
+    const tr: tr.ToolRunner = tl.tool('tar');
+    tr.arg(['xzC', destination, '-f', file]);
+    tr.exec();
+}
+
+/** Choose an extractor function based on the file type and agent's operating system. */
+function pickExtractor(fileType: CompressedFile, hostOS: string): Extractor {
+    const isWindows = hostOS === 'win32';
+
+    if (isWindows) {
+        switch (fileType) {
+            case CompressedFile.Tar:
```

Comment:
`Tar` uses the same compression as `SevenZip`?  If so, please provide a comment confirming that.

Reply:
NM, I see that most are using 7zip to do the actual work.